### DISCUSSION TITLE
Skin Editor: make PerformancePointsCounter expand left when anchored on the right

### DISCUSSION THREAD

**Discussion by Non-Binary-Programmer** [ppy/osu#23446] [Ideas] [answered]
![image](https://user-images.githubusercontent.com/100527514/236990062-e7f175c1-678f-40d0-96e2-2b58fdcc74d8.png)

In this image, I have PerformancePointsCounter anchored to the top right and displaying right under my accuracy. I have set it up so that it fits pleasingly in the top right corner. However, once I gain large amounts of performance points (2-3 digits worth), this happens:

![image](https://user-images.githubusercontent.com/100527514/236990537-9315532d-e716-4ea5-8609-c4a395f40355.png)

In this image, the text displayed by PerformancePointsCounter has increased in length (because the amount of performance points has increased), causing it to increase the size of the textbox to the right, making it misaligned with the other skin elements and cutting off the second 'p'.

To fix this, I would propose that the PerformancePointsCounter instead is able to expand the textbox to the left and keep the rightmost part of the box aligned. I propose that this would be the default behavior when the element is anchored to the right (TopRight, CenterRight, or BottomRight). This behavior could also be made into a setting of the PerformancePointsCounter.

**Comment by bdach** [comment, marked as answer]
This is already possible. Change origin to top right.

![1683608909](https://user-images.githubusercontent.com/20418176/236999392-5d68590a-da9d-42b3-8f2c-f4e4ce6a29cc.png)
